### PR TITLE
File hosting

### PR DESCRIPTION
**Pull request by alxdarksage**
Adding the metadata for a hosted file (for which there will be revisions).

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/services/FileServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.files.FileMetadata;
+
+public class FileServiceTest extends Mockito {
+
+    private static final String NAME = "oneName";
+    
+    @Spy
+    FileMetadataDao mockFileDao;
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/services/FileServiceTest.java:36]: Did you mean Mock? Is there any situation where we want to call through to the actual mockFileDao?

File: src/main/resources/db/changelog/changelog.sql [hunk 2]
```diff
@@ -208,3 +208,29 @@ CREATE TABLE IF NOT EXISTS `RequestInfos` (
...
+    `guid` varchar(60) NOT NULL,
+    `name` varchar(255) DEFAULT NULL,
+    `createdOn` BIGINT UNSIGNED NOT NULL,
+    `modifiedOn` BIGINT UNSIGNED NOT NULL,
+    `description` text,
+    `mimeType` varchar(255) DEFAULT NULL,
+    `deleted` tinyint(1) NOT NULL DEFAULT '0',
+    `version` int(10) unsigned NOT NULL,
+    PRIMARY KEY (`guid`),
+    KEY `Studies_idx` (`studyId`)
```
Comment (DwayneJengSage) [inline src/main/resources/db/changelog/changelog.sql]: Do we also want a compound index by studyId and guid?

**Review by DwayneJengSage — APPROVED** [review]
Looks good. Just a few minor questions.

File: src/test/java/org/sagebionetworks/bridge/services/FileServiceTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/services/FileServiceTest.java:36]: Yes this should be a mock.

File: src/main/resources/db/changelog/changelog.sql (hunk 2 above)
Comment (alxdarksage) [reply src/main/resources/db/changelog/changelog.sql]: Yes, we ensure study correctness by using both so we should add an index.

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**